### PR TITLE
pandas 2.2.0: Deprecated frequency alias `M` in favor of `ME`

### PR DESCRIPTION
**Pull request by hendrikmakait**
https://pandas.pydata.org/docs/whatsnew/v2.2.0.html#deprecate-aliases-m-q-y-etc-in-favour-of-me-qe-ye-etc-for-offsets

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by crusaderky** [comment]
- XREF https://github.com/dask/dask/pull/10577

**Review by crusaderky — CHANGES_REQUESTED** [review]
Test is still failing: 


E   FutureWarning: 'H' is deprecated and will be removed in a future version, please use 'h' instead.

File: distributed/tests/test_dask_collections.py [hunk 1]
```diff
@@ -139,12 +140,13 @@ async def test_bag_groupby_key_hashing(c, s, a, b):
 
 @pytest.mark.parametrize("wait", [wait, lambda x: None])
 def test_dataframe_set_index_sync(wait, client):
+    partition_freq_unit = "ME" if PANDAS_GE_220 else "M"
     df = dask.datasets.timeseries(
         start="2000",
         end="2001",
         dtypes={"value": float, "name": str, "id": int},
         freq="2H",
```
Comment (crusaderky) [inline distributed/tests/test_dask_collections.py:148]:
```suggestion
        freq="2h",
```

**Review by crusaderky — APPROVED** [review]

File: distributed/tests/test_dask_collections.py (hunk 1 above)
Comment (hendrikmakait) [reply distributed/tests/test_dask_collections.py]: Thanks!

**Pull request merged**